### ISSUE TITLE
Include the correct type into the property of the ContentBehavior

### ISSUE THREAD

**Issue by Anubarak** [craftcms/cms#3894]
### Description
First of all: The way you handle the ContentBehavior is absolutely awesome and I appreciate your idea really much. 
I'm not sure if you have ever thought about including a more specific type into the PHPDoc when you create this class? I know this is far away from being a first world problem but including a property like

```PHP
public static $phpDocType = 'mixed';
```
into `craft\base\Field` that can overwritten by any class that extends it. For example the `BaseRelationField` could have 

```PHP
public static $phpDocType = 'craft\elements\db\ElementQuery';
```
would be really awesome.

You could then fetch the type of the field together with the handle in your query and check if the specific class exists to grab the static attribute

```PHP
$docType = class_exists($fieldType)? $fieldType::$phpDocType : 'mixed';
//------
//------
/**
 * @var {$docType} Value for field with the handle “{$handle}”.
 */
public \${$handle};
```

If you think that might make Craft better but don't have the time for it I'm happy to create a pull request.

This way: when we type `$element->user->` PHPStorm (or others) won't complain about missmatches and suggest the functions of `ElementQuery`

**Comment by brandonkelly** [comment]
Yeah good idea. Feel free to PR it. Just keep in mind that the same field handle could be used by more than one field in different contexts (e.g. `body` could be a global field and a sub-field of a Matrix block type.) So you’d need to combine all the different property types together with `|`s.

**Comment by brandonkelly** [comment]
This is all merged into the `3.2` branch. Thanks! (Made several changes though—see 1c84c122e79ba14138168f0ce6f45ed42089d105)

**Comment by brandonkelly** [comment]
Added for 3.2 per your PR!
